routes/auth.routes.js

```diff
@@ -0,0 +1,85 @@
+const bcryptjs = require("bcryptjs");
+const saltRounds = 10;
+
+const mongoose = require("mongoose");
+const User = require("../models/User.model");
+
+const router = require("express").Router();
+
+const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard.js");
+
+//// SignUp ////
+
+router.get("/signup", isLoggedOut, (req, res, next) => {
+  res.render("auth/signup");
+});
+
+router.post("/signup", isLoggedOut, (req, res, next) => {
+  const { username, password } = req.body;
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({
+        username,
+        passwordHash: hashedPassword,
+      });
+    })
+    .then((user) => {
+      res.redirect("/userProfile");
```

Comment:
If you wish to redirect the user directly to the profile after the signup you should create a session for it before the redirection. Otherwise, the middleware will block it to see the route!

Like this
```   .then((user) => {
       req.session.currentUser = user;
        req.app.locals.currentUser = user;
      res.redirect("/userProfile");
}
```